e2e/utils/waits-utils.ts

```diff
@@ -388,20 +387,21 @@ export async function waitWhileLoading(
   await Promise.race([page.waitForSelector(notBlankPageSelector), page.waitForSelector(spinElementsSelector)]);
 
   // Wait for spinners stop and gone.
-  await page
-    .waitForFunction(
+  await Promise.all([
+    page.waitForFunction(
       (css) => {
-        const elements = document.querySelectorAll(css);
-        return elements && elements.length === 0;
+        return !document.querySelectorAll(css).length;
       },
       { polling: 'mutation', timeout },
       spinElementsSelector
-    )
-    .catch((err) => {
-      logger.error(`waitWhileLoading() failed: spinner xpath="${spinElementsSelector}"`);
-      logger.error(err);
-      throw new Error(err);
-    });
+    ),
+    page.waitForSelector(spinElementsSelector, { hidden: true, timeout })
```

Comment:
Why `waitWhileLoading` failed to work for workspace-card deletion is not clear to me. Adding a second check to see if it helps.